src/auth-service/utils/create-user.js

```diff
@@ -682,126 +634,74 @@ const join = {
             return {
               success: true,
               message: "forgot email successfully sent",
+              status: httpStatus.OK,
             };
           } else if (responseFromSendEmail.success === false) {
-            if (responseFromSendEmail.error) {
-              return {
-                success: false,
-                error: responseFromSendEmail.error,
-                message: "unable to send the email request",
-              };
-            } else {
-              return {
-                success: false,
-                message: responseFromSendEmail.message,
-              };
-            }
-          }
-        }
-
-        if (responseFromModifyUser.success === false) {
-          if (responseFromModifyUser.error) {
-            return {
-              success: false,
-              error: responseFromModifyUser.error,
-              message: responseFromModifyUser.message,
-            };
-          } else {
-            return {
-              success: false,
-              message: responseFromModifyUser.message,
-            };
+            return responseFromSendEmail;
           }
+        } else if (responseFromModifyUser.success === false) {
+          return responseFromModifyUser;
         }
       } else if (responseFromGenerateResetToken.success === false) {
-        if (responseFromGenerateResetToken.error) {
-          return {
-            success: false,
-            error: responseFromGenerateResetToken.error,
-            message: responseFromGenerateResetToken.message,
-          };
-        } else {
-          return {
-            success: false,
-            message: responseFromGenerateResetToken.message,
-          };
-        }
+        return responseFromGenerateResetToken;
       }
     } catch (e) {
       logElement("forgot password util", e.message);
       return {
         success: false,
-        message: "util server error",
+        message: "Internal Server Error",
         error: e.message,
+        errors: { message: e.message },
+        status: httpStatus.INTERNAL_SERVER_ERROR,
       };
     }
   },
 
-  updateForgottenPassword: async (tenant, filter, update) => {
+  updateForgottenPassword: async (request) => {
     try {
-      let responseFromCheckTokenValidity = await join.isPasswordTokenValid(
-        tenant.toLowerCase(),
-        filter
-      );
+      const { tenant, body } = request;
+      const { resetPasswordToken } = body;
+      const timeZone = moment.tz.guess();
+      let filter = {
+        resetPasswordToken,
+        resetPasswordExpires: {
+          $gt: moment().tz(timeZone).toDate(),
```

Comment:
this is the most recent addition to address timezone differences of the server which was the causing the datetime failure when updating forgotten passwords. Perhaps the recent changes in server settings led to this challenge which has been fixed with this module. @BenjaminSsempala @123MwanjeMike @Codebmk @NoahNsimbe